contentpal/src/test/java/org/dmfs/android/contentpal/utils/ContentOperationBuilderCountMatcher.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.android.contentpal.utils;
+
+import android.content.ContentProviderOperation;
+
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+
+import java.lang.reflect.Field;
+import java.util.Locale;
+
+
+/**
+ * A {@link ContentProviderOperation.Builder} {@link Matcher} which matches the operation type.
+ * <p>
+ * TODO: match Uri
+ *
+ * @author Marten Gajda
+ */
+public final class ContentOperationBuilderCountMatcher extends TypeSafeDiagnosingMatcher<ContentProviderOperation.Builder>
+{
+
+    private final Integer mExpectedCount;
+
+
+    public static ContentOperationBuilderCountMatcher expectingCount(int expectedCount)
+    {
+        return new ContentOperationBuilderCountMatcher(expectedCount);
+    }
+
+
+    public static ContentOperationBuilderCountMatcher expectingNoCount()
+    {
+        return new ContentOperationBuilderCountMatcher(null);
+    }
+
+
+    public ContentOperationBuilderCountMatcher(Integer expectedCount)
+    {
+        mExpectedCount = expectedCount;
+    }
+
+
+    @Override
+    protected boolean matchesSafely(ContentProviderOperation.Builder builder, Description mismatchDescription)
+    {
+        // create a builder an test the ContentValues
+        try
+        {
+            Field valuesField = ContentProviderOperation.Builder.class.getDeclaredField("mExpectedCount");
```

Comment:
For me, localized try-catches are always easier to read. I mean that when the `catch` branches are at the end of a longer code section I don't know where the exceptions are coming from. Could the retrieval of `mExpectedCount` field be extracted to a method here? That code also repeats over the 3 builder matchers, so it could also be reused.

Reply:
Something like this (using a `class FieldValue<T> implements Single<T>`?
```
Integer expectedCount = new FieldValue<Integer>(builder, "mExpectedCount").value();
```
That would be nice, but requires https://github.com/dmfs/jems/pull/6 to be merged an released first.